data/src/main/java/org/apache/iceberg/data/orc/GenericOrcWriters.java

```diff
@@ -231,7 +231,14 @@ public void nonNullWrite(int rowId, String data, ColumnVector output) {
 
     @Override
     public void nonNullWrite(int rowId, ByteBuffer data, ColumnVector output) {
-      ((BytesColumnVector) output).setRef(rowId, data.array(), 0, data.array().length);
+      ByteBuffer dupe = data.duplicate();
```

Comment:
There is no need to duplicate the `ByteBuffer` when accessing the backing on-heap byte array, so this should only be done in the case where we read from the `ByteBuffer` using `get`.